plugins/catalog-react/api-report.md

```diff
@@ -524,12 +524,24 @@ export type UnregisterEntityDialogProps = {
 };
 
 // @public
-export function useEntity<T extends Entity = Entity>(): {
-  entity: T;
+export function useAsyncEntity<
+  T extends Entity = Entity,
+>(): UseAsyncEntityResponse<T>;
+
+// @public
+export interface UseAsyncEntityResponse<T> {
+  // (undocumented)
+  entity?: T;
+  // (undocumented)
+  error?: Error;
+  // (undocumented)
   loading: boolean;
-  error: Error | undefined;
-  refresh: VoidFunction | undefined;
-};
+  // (undocumented)
+  refresh?: VoidFunction;
+}
+
+// @public
+export function useEntity<T extends Entity = Entity>(): UseEntityResponse<T>;
```

Comment:
I think we should just inline `UseEntityResponse` instead tbh, it's gonna be shorter anyway 😅

Naming-wise I think response is typically for async request responses, and `*Result` perhaps would've been better here, but I say we just avoid that debate x)